srcs/utils/free_memory.c

```diff
@@ -50,6 +50,10 @@ void	free_table(t_cmd_table *cmd_table, int nb_cmds)
 			free_split(cmd_table[i].flags);
 		if (cmd_table[i].cmd_args)
 			free_split(cmd_table[i].cmd_args);
+		if (cmd_table[i].fd_in != 0)

```

Comment:
I'm confused by that.
I think you're supposed to close those in the pipes already no ?

Reply:
I do, but only when things go well or when I manage the errors myself so to speak. When I call "something_or_exit" I don't manage it if it exits. It might happen that we call close on the same fd twice, you're right about that, but close will simply return -1 and fail silently (unless we do something about it, which I don't, on purpose)